packages/cloudflare/src/index.ts

```diff
@@ -45,19 +42,24 @@ type Options = {
 		/** Additional `exclude` patterns */
 		exclude?: string[];
 	};
+	// TODO: rewrite this inline code commnent to make more sense
 	/**
-	 * Going forward only the object API should be used. The modes work as known before:
-	 * 'off': current behaviour (wrangler is needed)
-	 * 'local': use a static req.cf object, and env vars defined in wrangler.toml & .dev.vars (astro dev is enough)
-	 * 'remote': use a dynamic real-live req.cf object, and env vars defined in wrangler.toml & .dev.vars (astro dev is enough)
+	 * { mode: 'off' }: current behaviour (wrangler is needed)
+	 * { mode: 'local', ... }: adds cf request object, locals bindings, env vars/secrets which are defined by the user to `astro.dev` with `Astro.locals.runtime` / `context.locals.runtime`
 	 */
 	runtime?:
-		| 'off'
-		| 'local'
-		| 'remote'
 		| { mode: 'off' }
-		| { mode: 'remote' }
-		| { mode: 'local'; persistTo?: string };
+		| {
+				mode: 'local';
+				type: 'pages';
+				persistTo?: string;
+				bindings?: Extract<RUNTIME, { type: 'pages' }>['bindings'];
+		  }
+		| {
+				mode: 'local';
+				type: 'workers';
+				persistTo?: string;
+		  };
```

Comment:
`'remote' | { mode: 'remote' }` were non-functional / undocumented, so it is not an issue to remove them in this major directly